tests/test_benchmarks_http_websocket.py

```diff
@@ -0,0 +1,63 @@
+"""codspeed benchmarks for http websocket."""
+
+import asyncio
+
+from pytest_codspeed import BenchmarkFixture  # type: ignore[import-untyped]
+
+from aiohttp import DataQueue
+from aiohttp.http_websocket import (
+    WebSocketReader,
+    WebSocketWriter,
+    WSMessage,
+    WSMsgType,
+)
+
+
+def test_read_one_hundred_websocket_text_messages(
+    loop: asyncio.AbstractEventLoop, benchmark: BenchmarkFixture
+) -> None:
+    queue: DataQueue[WSMessage] = DataQueue(loop=loop)
+    reader = WebSocketReader(queue, max_msg_size=2**16)
+    raw_message = (
+        b'\x81~\x01!{"id":1,"src":"shellyplugus-c049ef8c30e4","dst":"aios-1453812500'
+        b'8","result":{"name":null,"id":"shellyplugus-c049ef8c30e4","mac":"C049EF8C30E'
+        b'4","slot":1,"model":"SNPL-00116US","gen":2,"fw_id":"20231219-133953/1.1.0-g3'
+        b'4b5d4f","ver":"1.1.0","app":"PlugUS","auth_en":false,"auth_domain":null}}'
+    )
+    feed_data = reader.feed_data
+
+    @benchmark
+    def _run() -> None:
+        for _ in range(100):
+            feed_data(raw_message)
+
+
+def test_send_one_hundred_websocket_text_messages(
+    loop: asyncio.AbstractEventLoop, benchmark: BenchmarkFixture
+) -> None:
+
+    class MockTransport(asyncio.Transport):
+        """Mock transport for testing that do no real I/O."""
+
+        def is_closing(self) -> bool:
+            """Swallow is_closing."""
+            return False
+
+        def write(self, data: bytes) -> None:
+            """Swallow writes."""
+
+    class MockProtocol(asyncio.Protocol):
+
+        async def _drain_helper(self) -> None:
+            """Swallow drain."""
+
+    writer = WebSocketWriter(MockProtocol(), MockTransport())
+    raw_message = b"Hello, World!" * 100
+
+    async def _send_one_hundred_websocket_text_messages() -> None:
+        for _ in range(100):
+            await writer.send_frame(raw_message, WSMsgType.TEXT, compress=False)
```

Comment:
```suggestion
            await writer.send_frame(raw_message, WSMsgType.TEXT)
```

Let's not benchmark kwarg overhead